src/visidentifier/visidentifier.cpp

```diff
@@ -0,0 +1,416 @@
+            if (auto err = newSubjects.PushBack(subject.c_str()); !err.IsNone()) {
+                return err;
+            }
+        }
+
+        std::lock_guard lock(mMutex);
+
+        if (mSubjects != newSubjects) {
+            mSubjects = std::move(newSubjects);
+            mSubjectsObserverPtr->SubjectsChanged(mSubjects);
+        }
+    } catch (const Poco::Exception& e) {
+        LOG_ERR() << e.message().c_str();
+
+        return aos::ErrorEnum::eFailed;
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+std::string VISIdentifier::SendGetRequest(const std::string& path)
+{
+    const auto       requestId = mWsClientPtr->GenerateRequestID();
+    const VISMessage getMessage(VISActionEnum::eGet, requestId, path);
+
+    WaitUntilConnected();
+
+    const auto response = mWsClientPtr->SendRequest(requestId, getMessage.ToByteArray());
+
+    return {response.cbegin(), response.cend()};
+}
+
+void VISIdentifier::SendUnsubscribeAllRequest()
+{
+    try {
+        const VISMessage request(VISActionEnum::eUnsubscribeAll, mWsClientPtr->GenerateRequestID(), "");
+
+        mWsClientPtr->AsyncSendMessage(request.ToByteArray());
+
+    } catch (const AosException& e) {
+        LOG_ERR() << e.what();
+    }
+}
+
+void VISIdentifier::Subscribe(const std::string& path, VISSubscriptions::Handler&& callback)
+{
+    const auto       requestId = mWsClientPtr->GenerateRequestID();
+    const VISMessage subscribeMessage(VISActionEnum::eSubscribe, requestId, path);
+
+    const auto       response = mWsClientPtr->SendRequest(requestId, subscribeMessage.ToByteArray());
+    const VISMessage responseVISMessage(std::string {response.cbegin(), response.cend()});
+
+    mSubscriptions.RegisterSubscription(
+        responseVISMessage.GetValue<std::string>(cSubscriptionIdTagName), std::move(callback));
+}
+
+std::string VISIdentifier::GetValueByPath(Poco::Dynamic::Var object, const std::string& valueChildTagName)
+{
+    auto var = UtilsJson::FindByPath(object, {cValueTagName});
+
+    if (var.isString()) {
+        return var.extract<std::string>();
+    }
+
+    var = UtilsJson::FindByPath(var, {valueChildTagName});
+
+    return var.extract<std::string>();
+}
+
+std::vector<std::string> VISIdentifier::GetValueArrayByPath(
+    Poco::Dynamic::Var object, const std::string& valueChildTagName)
+{
+    auto var = UtilsJson::FindByPath(object, {cValueTagName});
+
+    const auto isArray = [](const Poco::Dynamic::Var var) {
+        return var.type() == typeid(Poco::JSON::Array) || var.type() == typeid(Poco::JSON::Array::Ptr);
+    };
+
+    if (!isArray(var)) {
+        var = UtilsJson::FindByPath(var, {valueChildTagName});
+    }
+
+    Poco::JSON::Array::Ptr array;
+    if (var.type() == typeid(Poco::JSON::Array::Ptr)) {
+        array = var.extract<Poco::JSON::Array::Ptr>();
+    } else if (var.type() == typeid(Poco::JSON::Array)) {
+        array = new Poco::JSON::Array(var.extract<Poco::JSON::Array>());
```

Comment:
if we allocate memory i think we should deallocate it

Reply:
It's stored in Array::Ptr
What's actually a smart pointer.